crates/ide_assists/src/handlers/move_module_to_file.rs

```diff
@@ -58,9 +58,18 @@ pub(crate) fn move_module_to_file(acc: &mut Assists, ctx: &AssistContext) -> Opt
                 }
                 let segments = iter::successors(Some(module_ast.clone()), |module| module.parent())
                     .filter_map(|it| it.name())
+                    .map(|name| SmolStr::from(name.text().trim_start_matches("r#")))
```

Comment:
I'm not sure if using `SmolStr` is appropriate here. I wanted to avoid copying strings but couldn't find any way to do so (correct me if I'm wrong but `SmolStr::from()` copies the source string right?).

Practically I don't think it'd cause major performance issues, though, as module names are usually not that long.

Reply:
Fwiw, SmolStr also allocates if the string is bigger than 15 characters, otherwise it's inline.

And yes, performance here does not matter that much since this is only executed when the assist is being executed.